src/db/index.ts

```diff
@@ -73,3 +91,24 @@ export async function saveCommit(commitInfo: CommitInfo): Promise<void> {
     );
   });
 }
+
+export interface CommitWithDiagnoses extends Commit {
+  diagnoses: CommitDiagnosis[];
+}
+
+export async function getTopCommitsWithDiagnoses(
+  afterDate: Date = new Date(0),
+  limitCount = 10
+): Promise<CommitWithDiagnoses[]> {
+  const commits = await getTopCommitsAfterDate(afterDate, limitCount);
+  const commitsWithDiagnoses = await Promise.all(
```

Comment:
This reproduces the traditional [N+1 problem](https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping) of ORM. Can we use a join instead?